rmgpy/molecule/molecule.py

```diff
@@ -1618,4 +1618,21 @@ def toGroup(self):
         
         return group
 
-
+    def getNthNeighbor(self, startingAtoms, n, ignoreList = []):
+        '''
+        Recursively get the Nth nonHydrogen neighbors of the startingAtoms, and return them in a list.
+        `startingAtoms` is a list of :class:Atom for which we will get the nth neighbor.
+        `n` is an interger.
+        `ignoreList` is a list of :class:Atom that have been counted in (n-1)th neighbor, and will not be returned.
+        '''
+        neighbors = []
+        for atom in startingAtoms:
+            newNeighbors = [neighbor for neighbor in self.getBonds(atom) if neighbor.isNonHydrogen()]
+            neighbors.extend(newNeighbors)
+
+        neighbors = list(set(neighbors)-set(ignoreList))
+        for atom in startingAtoms:
+            ignoreList.append(atom)
+        if n>1:
+            neighbors = self.getNthNeighbor(neighbors, n-1, ignoreList)
+        return neighbors
```

Comment:
I removed the confusing getBaseCb method in 32705582bb4d1adcaab97fd61cb10f79c360ff7a since we don't need it anymore.